### ISSUE TITLE
Feature request: give new parameter to a tf.summary.scalar to indicate which graph it belongs to

### ISSUE THREAD

**Issue by aforslow** [tensorflow/tensorboard#969]
I have stumbled upon the difficulties of plotting more plots into the same graph using tf.summary.scalar during one session run. I, like many others, want to plot a train and validation curve on the same graph, but have only seen workarounds to this, where two different filewriters create two different directories for saving the two different plots. This is indeed a workaround, and sometimes it does not even work - for example when I want to save a tf.summary.scalar with the same name for train and validation set during one run. Another example that this workaround doesn't work on, as I saw someone else point out in issue #300 , is the A3C algorithm.

I have read this thread and seen that there is a tensorboard plugin (tensorboard/plugins/custom_scalar) that also is a workaround to the issue; a plugin that I, unfortunately, haven't managed to get to work properly. I wanted to mention this to show that I am aware that this plugin exists, and to point out that I am looking for a different solution.

The solution I want is to make it possible to provide a parameter to tf.summary.scalar to indicate which graph it belongs to. The tf.summary.scalar would then have both a unique identifier in the name parameter, along with a graph identifier in the graph_name parameter. The complete function would then look like this:

tf.summary.scalar(name, tensor, graph_name, collections, family)

Is this possible?

**Comment by nfelt** [comment]
Can you elaborate more on what you couldn't get to work with the custom scalars plugin?  We built it to solve just this kind of problem and it's ultimately much more powerful than just being able to specify a single graph_name on a scalar summary, since e.g. if you wanted to plot the same scalar value on two different charts then that already wouldn't work for that use case.

**Comment by aforslow** [comment]
Yeah, that's true, even though that particular situation could probably be solved with graph_name being a list instead  of single value. The reason I want this feature is that the coding becomes very simple with only one extra parameter passed to tf.summary.scalar when building the graph. Very intuitive if you just want to quickly graph two plots against each other in the same run. Perhaps this could work as an extension to the custom scalars plugin, for someone who just wants to plot two scalar curves against each other and not care that much about how it is done?

As for the custom scalars plugin, I realized earlier today that I missed noticing the custom scalars tab, so it works fine! I've also started using it, and like it quite much after having played around with it a bit, so thanks for developing this. That said, though, it feels like quite a bit of code in order to get two plots into one graph. Would it be possible to make an extension of the plugin as mentioned above?

**Comment by nfelt** [comment]
I'm glad to hear you're able to use the custom scalars plugin and like it so far!

Re: the original request, I think for now at least our answer is going to be that the custom scalars plugin is the right tool for this, even if it's a bit heavyweight for the simplest case of plotting two scalar tags on the same chart.  Changing `tf.summary.scalar()` would require a fairly high bar since it's so widely used (even extra optional parameters can possibly lead to user confusion), and it would introduce duplication of functionality with respect to the custom scalars plugin, while ultimately not being able to support the full-featured configuration that the custom scalars plugin provides.

I think if we did want improve the "plot two scalar curves against each other" case, the way we'd likely want to do that is by adding "shortcut" methods for using the custom scalars plugin, e.g. some slightly friendlier API for building the appropriate Layout protocol buffer object, where we could say for example add a `tb.summary.custom_scalar_multiline(title, tag_list)` helper method.  That way it preserves the concept of specifying the tags for a given plot, rather than the plots for a given tag, which matches the way we've structured the custom scalars dashboard on the whole.  But I'll defer to @chihuahua on whether something like that might be appropriate, especially since it might require changes to e.g. merge layouts within a run as well as between runs, etc.
